akd/src/proto/audit.proto

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+// This contains the protobuf definition for inter-node messaging structures
+
+// Note: install a proper version of protobuf-codegen before re-generating
+// protobuf-codegen --version 2.8.1
+// Instructions: https://github.com/stepancheg/rust-protobuf/tree/master/protobuf-codegen
+// MAKE SURE to cargo install the right version: cargo install protobuf-codegen --version =2.8.1
+
+// Command to re-generate (from 'proto' directory):
+// protoc --rust_out . audit.proto
+
+syntax = "proto2";
+
+/* NodeLabel represents the label of a history tree node in the AKD tree with a
+ * supplied label-length and label value (location) */
+message NodeLabel {
+    optional bytes label_val = 1;
+    optional uint32 label_len = 2;
```

Comment:
So I know that `optional` is the recommended way of adding fields to messages for compatibility reasons, but do we really foresee these ever being removed? That is, is it possible that we can mark these as required? I ask because we can potentially eliminate the macro we wrote for ensuring that fields are present and instead rely on proto to handle that for us.

Reply:
So from the protobuf guide:

> Required Is Forever You should be very careful about marking fields as required. If at some point you wish to stop writing or sending a required field, it will be problematic to change the field to an optional field – old readers will consider messages without this field to be incomplete and may reject or drop them unintentionally. You should consider writing application-specific custom validation routines for your buffers instead.
> 
> A second issue with required fields appears when someone adds a value to an enum. In this case, the unrecognized enum value is treated as if it were missing, which also causes the required value check to fail.

At this point, the macros are there, they work as expected, I think it's better to leave them if we do ever break something in the future. I'd much prefer as *much* backwards compatibility and flexibility as possible

Reply:
Works for me! Figured I'd ask to spark the conversation! I think you make good points and am good with leaving as much backwards compatibility/flexibility as possible at this point.